sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -950,21 +950,31 @@ CURLcode CurlSession::ReadStatusLineAndHeadersFromRawResponse(
     // The value can also be "keep-alive, close" (i.e. "both are fine"), in which case we are
     // preferring to treat it as keep-alive.
     // (https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Connection)
-    // Should it come to HTTP/2 and HTTP/3, they are "keep-alive", but any response from HTTP/2 or
-    // /3 containing a "Connection" header should be considered malformed. (HTTP/2:
-    // https://httpwg.org/specs/rfc9113.html#ConnectionSpecific
-    //  HTTP/3: https://httpwg.org/specs/rfc9114.html#rfc.section.4.2)
-    if (m_response->GetMajorVersion() == 1 && m_response->GetMinorVersion() >= 1)
+    if (m_response->GetMajorVersion() == 1)
     {
-      m_httpKeepAlive = (!hasConnectionClose || hasConnectionKeepAlive);
-    }
-    else if (m_response->GetMajorVersion() <= 1)
-    {
-      m_httpKeepAlive = hasConnectionKeepAlive;
+      if (m_response->GetMinorVersion() >= 1)
+      {
+        // HTTP/1.1+
+        m_httpKeepAlive = (!hasConnectionClose || hasConnectionKeepAlive);
+      }
+      else
+      {
+        // HTTP/1.0
+        m_httpKeepAlive = hasConnectionKeepAlive;
+      }
     }
     else
     {
-      m_httpKeepAlive = true;
+      Log::Write(
+          Logger::Level::Verbose,
+          LogMsgPrefix
+              + "Unsupported HTTP version in the response. Only HTTP/1.0 and HTTP/1.1 is "
+                "supported as a response, for an HTTP/1.1 request.");
```

Comment:
Instead of this message, let's update logging policy to log HTTP version for responses, and, optionally, for requests. The `Connection` header is already in the `g_defaultAllowedHttpHeaders`, so it should be sufficient info for diagnistic purposes - everything in a single log entry, plus it will cover both WinHTTP and libcurl adapters.

Reply:
It's marked as verbose, I don't see any harm in keeping the log message here, as something exceptional has happened, even if it is specific to curl.

> Let's update logging policy to log HTTP version for responses, and, optionally, for requests

Can you please file an issue for that. It seems reasonable to do, imo, but not as part of this change.

Reply:
FWIW, HTTP Requests *must* always be HTTP 1.1, as there are a more than trivial number of Azure services which will reject request that aren't on HTTP 1.1 :).

Reply:
I know, but logging policy doesn't have to know that our transports do hardcode 1.1 for requests :)